Applications/Utils/ModelPreparation/PartitionMesh/NodeWiseMeshPartitioner.cpp

```diff
@@ -267,27 +279,54 @@ bool copyCellPropertyVector(MeshLib::Properties const& original_properties,
         name, pv->getMeshItemType(), pv->getNumberOfComponents());
     partitioned_pv->resize(total_number_of_tuples *
                            pv->getNumberOfComponents());
-    std::size_t position_offset(0);
-    for (auto const& p : partitions)
-    {
-        std::size_t const n_regular(p.regular_elements.size());
-        for (std::size_t i = 0; i < n_regular; ++i)
+
+    auto copy_property_vector_values = [&](Partition const& p,
+                                           std::size_t offset) {
+        if (pv->getMeshItemType() == MeshLib::MeshItemType::Node)
         {
-            const auto id = p.regular_elements[i]->getID();
-            (*partitioned_pv)[position_offset + i] = (*pv)[id];
+            return copyNodePropertyVectorValues(p, offset, *pv,
+                                                *partitioned_pv);
         }
-        position_offset += n_regular;
-        std::size_t const n_ghost(p.ghost_elements.size());
-        for (std::size_t i = 0; i < n_ghost; ++i)
+        if (pv->getMeshItemType() == MeshLib::MeshItemType::Cell)
         {
-            const auto id = p.ghost_elements[i]->getID();
-            (*partitioned_pv)[position_offset + i] = (*pv)[id];
+            return copyCellPropertyVectorValues(p, offset, *pv,
+                                                *partitioned_pv);
         }
-        position_offset += n_ghost;
+        OGS_FATAL(
+            "Copying of property vector values for mesh item type %s is "
+            "not implemented.",
+            pv->getMeshItemType());
+    };
+
+    std::size_t position_offset(0);
+    for (auto p : partitions)
+    {
+        position_offset += copy_property_vector_values(p, position_offset);
     }
     return true;
 }
 
+template <typename Function>
```

Comment:
IMHO this code would be easier to understand if you would provide the function `copy_property_vector` right here abov this line and not pass it as a lambda argument.